pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -87,9 +119,122 @@ func (pl *VolumeBinding) Filter(ctx context.Context, cs *framework.CycleState, p
 	return nil
 }
 
-// New initializes a new plugin with volume binder and returns it.
-func New(_ runtime.Object, fh framework.FrameworkHandle) (framework.Plugin, error) {
+// Reserve reserves volumes of pod and saves binding status in cycle state.
+func (pl *VolumeBinding) Reserve(ctx context.Context, cs *framework.CycleState, pod *v1.Pod, nodeName string) *framework.Status {
+	allBound, err := pl.binder.AssumePodVolumes(pod, nodeName)
+	if err != nil {
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+	cs.Write(allBoundStateKey, stateData{allBound: allBound})
+	return nil
+}
+
+// PreBind will make the API update with the assumed bindings and wait until
+// the PV controller has completely finished the binding operation.
+//
+// If binding errors, times out or gets undone, then an error will be returned to
+// retry scheduling.
+func (pl *VolumeBinding) PreBind(ctx context.Context, cs *framework.CycleState, pod *v1.Pod, nodeName string) *framework.Status {
+	state, err := cs.Read(allBoundStateKey)
+	if err != nil {
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+	s, ok := state.(stateData)
+	if !ok {
+		return framework.NewStatus(framework.Error, "unable to convert state into stateData")
+	}
+	if s.allBound {
+		// no need to bind volumes
+		return nil
+	}
+	klog.V(5).Infof("Trying to bind volumes for pod \"%v/%v\"", pod.Namespace, pod.Name)
+	err = pl.binder.BindPodVolumes(pod)
+	if err != nil {
+		klog.V(1).Infof("Failed to bind volumes for pod \"%v/%v\": %v", pod.Namespace, pod.Name, err)
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+	klog.V(5).Infof("Success binding volumes for pod \"%v/%v\"", pod.Namespace, pod.Name)
+	return nil
+}
+
+// Unreserve clears pod binding state.
+// TODO(cofyc) Revert assumed PV/PVC cache, see http://issues.k8s.io/82934#issuecomment-538269188
+func (pl *VolumeBinding) Unreserve(ctx context.Context, cs *framework.CycleState, pod *v1.Pod, nodeName string) {
+	pl.binder.DeletePodBindings(pod)
+	return
+}
+
+// PostBind is called after a pod is successfully bound.
+func (pl *VolumeBinding) PostBind(ctx context.Context, cs *framework.CycleState, pod *v1.Pod, nodeName string) {
+	pl.binder.DeletePodBindings(pod)
+	return
+}
+
+// New initializes a new plugin and returns it.
+func New(plArgs runtime.Object, fh framework.FrameworkHandle) (framework.Plugin, error) {
+	var args *schedulerv1alpha2.VolumeBindingArgs
+	args, ok := plArgs.(*schedulerv1alpha2.VolumeBindingArgs)
+	if !ok {
+		args = &schedulerv1alpha2.VolumeBindingArgs{}
+		if err := framework.DecodeInto(plArgs, args); err != nil {
+			return nil, err
+		}
+	}
+	if err := validateArgs(args); err != nil {
+		return nil, err
+	}
+	bindTimeoutSeconds := int64(DefaultBindTimeoutSeconds)
+	if args.BindTimeoutSeconds != nil {
```

Comment:
This can go into `pkg/scheduler/apis/config/v1alpha2/defaults.go`

This is something rather new. Sorry for the back-and-forth. The whole thing about plugin args was a moving target, but we are finally settling (#90657)